### PR TITLE
Added improvements in resolving the best route

### LINKED ISSUE #15762: Weird routes generated for multi tx
There is 
0.03 eth on mainnet
0.03 eth on optimism
0.01 eth on arbitrum

When user tries to send 0.04, when all "from" and "to" networks are enabled, too many routes get generated.

_a1

### PR DESCRIPTION
**Pull request by saledjenic**
This PR adds improvements in resolving the best route for multi tx, should fix https://github.com/status-im/status-desktop/issues/15762

Some new params are also exposed in this PR for convenience:
- `MaxFeesPerGas                   *hexutil.Big   // Max fees per gas (determined by client via GasFeeMode, in ETH WEI)`
- `TxFee   *hexutil.Big // fee for the transaction (includes tx fee only, doesn't include approval fees, l1 fees, l1 approval fees, token fees or bonders fees, in ETH WEI)`
- `ApprovalFee   *hexutil.Big // Total fee for the approval transaction (includes approval tx fees only, doesn't include approval l1 fees, in ETH WEI)`
- `TxTotalFee *hexutil.Big // Total fee for the transaction (includes tx fees, approval fees, l1 fees, l1 approval fees, in ETH WEI)`

This is the issue we had:
_a1

And with these changes it looks like this:
_a2

### CODE REVIEW

File: services/wallet/router/fees.go [hunk 1]
```diff
@@ -54,27 +54,23 @@ type SuggestedFeesGwei struct {
 	EIP1559Enabled       bool       `json:"eip1559Enabled"`
 }
 
-func (s *SuggestedFees) feeFor(mode GasFeeMode) *big.Int {
-	if !s.EIP1559Enabled {
```
Comment (IvanBelyakoff) [inline services/wallet/router/fees.go:58]: May I ask you why it is removed?

**Review by Cuteivist — APPROVED** [review]
LGTM

**Review by IvanBelyakoff — APPROVED** [review]
LGTM. Could you pleas answer my inline question?

**Review by belalshehab — APPROVED** [review]

File: services/wallet/router/fees.go (hunk 1 above)
Comment (saledjenic) [reply services/wallet/router/fees.go:58]: Because of implementation simplifications, look at all changes done in that file.
Lines 140-142.

**Comment by J-Son89** [comment]
can you please add a link in the description to the issue this is solving 👍

**Comment by saledjenic** [comment]
@J-Son89 added. :)

File: services/wallet/router/fees.go [hunk 2]
```diff
@@ -141,9 +137,9 @@ func (f *FeeManager) SuggestedFees(ctx context.Context, chainID uint64) (*Sugges
 			BaseFee:              big.NewInt(0),
 			MaxPriorityFeePerGas: big.NewInt(0),
 			MaxFeesLevels: &MaxFeesLevels{
-				Low:    (*hexutil.Big)(big.NewInt(0)),
-				Medium: (*hexutil.Big)(big.NewInt(0)),
-				High:   (*hexutil.Big)(big.NewInt(0)),
+				Low:    (*hexutil.Big)(gasPrice),
+				Medium: (*hexutil.Big)(gasPrice),
+				High:   (*hexutil.Big)(gasPrice),
```
Comment (alaibe) [inline services/wallet/router/fees.go:142]: that used to be there to detect non eip1559 chain but there is no such chain anymore right?

Shall we return the error?

Comment (saledjenic) [reply services/wallet/router/fees.go:142]: Everything is the same, I added `func (m *MaxFeesLevels) feeFor(mode GasFeeMode) *big.Int`, that way we're getting appropriate MaxFee based on the passed `GasFeeMode`, then I updated `func (s *SuggestedFees) feeFor(mode GasFeeMode) *big.Int` which previously had `if !s.EIP1559Enabled` condition, under which it was returning `GasPrice`, but now in the case of `EIP1559Enabled` we're returning default `MaxFee` of `MaxFeesLevels ` which is set to `GasPrice`, so basically the effect is the same, just written differently. We can revert this part and nothing will be changed.

**Pull request merged**